pkg/proto/nbcontract/v1/helper.go

```diff
@@ -93,9 +91,6 @@ func (nBCB *NBContractBuilder) deepCopy(src, dst interface{}) error {
 // It returns an error if the contract is invalid.
 // This function should be called after applying all configuration and before sending to downstream component.
 func (nBCB *NBContractBuilder) ValidateNBContract() error {
-	if err := nBCB.validateSemVer(); err != nil {
```

Comment:
Why remove this? This is to check if the contract version is valid.

Reply:
Two points:

1) We should avoid using semantic versioning (semver) for config versioning. I don't anticipate any functional difference between, for example, config versions 1.0.0 and 1.0.1. Requiring users to constantly update versions for minor or patch-level changes adds unnecessary complexity.
2) We only need to track major version changes. Different versions will correspond to different structs, meaning version parsing and validation will take place before unmarshalling the config. Currently, the only supported version is v0 (this allows us to introduce breaking changes before the final release if needed; if no changes are required, v0 will effectively serve as a synonym for v1).

Reply:
1. User doesn't need to update the version manually. If he/she is using the nbcContractBuilder, it will apply the version from the package.
2. Right. Only major version matters in this validation function. Minor version just returns a warning. So why removing the major version check?

Reply:
I don't think we should force users to use nbcContractBuild to build the config.

https://github.com/Azure/AgentBaker/pull/5137/files#diff-54d0a4d20361aa9831f61f6129558f7567bd116eb245ca57ac21ff4de4d40cd5R40-R41

I moved the validation is there. In the future I think we will have logic like this:
```
1) Parse json. But read only `version` property.
2) If `version == 0`, then parse json into ContractV0 struct, if `version==1` then parse json into ContractV1 struct, otherwise fail.
```
Currently we support only single version, so the logic isn't implemented yet.

Reply:
Agree. We don't need to force user to use the builder. Also agree we probably don't need semver for contract version otherwise we will need to keep updating the version when we add a new field. If we already have a place to check the version, I am fine with it.